moveit_simple/src/moveit_simple.cpp

```diff
@@ -1232,16 +1281,34 @@ bool Robot::getIK(const Eigen::Affine3d pose, const std::vector<double> & seed,
 bool Robot::getIK(const Eigen::Affine3d pose, std::vector<double> & joint_point,
                   double timeout, unsigned int attempts) const
 {
-  if ( virtual_robot_state_->setFromIK(joint_group_, pose, attempts, timeout) )
-  {
-    virtual_robot_state_->copyJointGroupPositions(joint_group_->getName(), joint_point);
-    virtual_robot_state_->update();
-    virtual_visual_tools_->deleteAllMarkers();
-    virtual_visual_tools_->publishRobotState(virtual_robot_state_, rviz_visual_tools::PURPLE);
-    virtual_visual_tools_->publishContactPoints(*virtual_robot_state_, &(*planning_scene_));  
-    ros::spinOnce();
-    return true;
+  if (ik_transforms_found_)
+  {
+    Eigen::Affine3d tool0_pose = pose * tool0_to_tcp_;
+    if (virtual_robot_state_->setFromIK(joint_group_, tool0_pose, attempts, timeout))
+    {
+      virtual_robot_state_->copyJointGroupPositions(joint_group_->getName(), joint_point);
+      virtual_robot_state_->update();
+      virtual_visual_tools_->deleteAllMarkers();
+      virtual_visual_tools_->publishRobotState(virtual_robot_state_, rviz_visual_tools::PURPLE);
+      virtual_visual_tools_->publishContactPoints(*virtual_robot_state_, &(*planning_scene_));  
+      ros::spinOnce();
+      return true;      
+    }
+  }
+  else
+  {
+    if (virtual_robot_state_->setFromIK(joint_group_, pose, attempts, timeout))
```

Comment:
This is a direct copy from above.  Can it be written better?

Reply:
Yes.